tests/post/features/volume.feature

```diff
@@ -148,3 +148,20 @@ Feature: Volume management
         And I delete the Volume 'test-volume10'
         Then the Volume 'test-volume10' does not exist
         And the PersistentVolume 'test-volume10' does not exist
+
+    Scenario: Test deletion while creation is in progress
+        Given the Kubernetes API is available
+        When I create the following Volume:
+            apiVersion: storage.metalk8s.scality.com/v1alpha1
+            kind: Volume
+            metadata:
+              name: test-volume11
+            spec:
+              nodeName: bootstrap
+              storageClassName: metalk8s-prometheus
+              sparseLoopDevice:
+                size: 10Gi
+        Then the Volume 'test-volume11' is 'Pending'
+        When I delete the Volume 'test-volume11'
+        Then the Volume 'test-volume11' does not exist
```

Comment:
Are we sure this is going to test "deletion while creation"?

It's the same TOCTOU as you often remember me about :) Sure, volume was `Pending`, but when you hit the `delete Volume` line, maybe it's not `Pending` anymore...

Maybe a single `When I create and immediately delete the following Volume: ...` would help.

Reply:
Yeah, it's racy but I trust Salt to be sloooooow enough to work (if it's flaky the test will succeed anyway so it won't bother us too much).

Having a single step may help but would still be racy (I had the case when testing manually where the delete would happen fast enough that the volume was still in "no state", not yet in Pending)